.github/workflows/main.yaml

```diff
@@ -0,0 +1,46 @@
+name: "nabula build"
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      id-token: write
+    steps:
+      - uses: actions/checkout@v3
+      - name: prepare
+        run: |
+          docker compose up -d --build
+          docker compose exec -T vault-secrets-operator git config --global url."https://saltosystems-bot:${{ secrets.SALTOSYSTEMS_BOT_GITHUB_TOKEN }}@github.com/".insteadOf "https://github.com/"
+      - name: sanity-check
+        run: |
+          docker compose exec -T vault-secrets-operator make check-fmt
+          docker compose exec -T vault-secrets-operator go mod tidy
+      - name: build
+        run: docker compose exec -T vault-secrets-operator make build docker-build
+      - id: auth
+        name: authentication
+        if: github.ref == 'refs/heads/main'
```

Comment:
Let's replace this to be just authenticated and released when a tag is set. The idea is to have this: `europe-docker.pkg.dev/salto-artifacts/private-docker/vault-secrets-operator:v0.3.4`